src/LocalDatastore.js

```diff
@@ -264,15 +261,16 @@ const LocalDatastore = {
         if (pinned.includes(objectKey)) {
           pinned = pinned.filter(item => item !== objectKey);
           if (pinned.length == 0) {
-            await this.unPinWithName(key);
+            promises.push(this.unPinWithName(key));
```

Comment:
didn't catch this in the first pass, but yeah, if you have an await in a loop...

our lint in parse-server will complain.